helfi_features/helfi_toc/assets/js/tableOfContents.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+(function($, Drupal) {
+  Drupal.behaviors.table_of_contents = {
+    attach: function attach() {
+      var anchorLinks = [];
+      var tableOfContents = $('#helfi-toc-table-of-contents');
+      var tableOfContentsList = $('#helfi-toc-table-of-contents-list > ul');
+      var mainContent = $('main.layout-main-wrapper');
+
+      // Do not include sidebar H2 or Table of contents H2.
+      var titleComponents = [
+        'h2:not(aside *)' +
+        ':not(.unit__sidebar *)' +
+        ':not(.service__sidebar *)' +
+        ':not(#helfi-toc-table-of-contents *)'
+      ];
+
+      // Craft table of contents.
+      $(titleComponents.join(','), mainContent)
+        .once()
+        .each(function(index) {
+          var anchorName =
+            index +
```

Comment:
I do not think that this is wanted. I think we should have a better approach that does not require the table of contents to stay in exact same order.

This is what current code produces:

* 0-the-first-heading
* 1-example-heading
* 2-nice-to-know
* 3-example-heading

This would break the links when the nice-to-know heading moves below the next one

* 0-the-first-heading
* 1-example-heading
* 2-example-heading (previous link here was 3-example-heading)
* 3-nice-to-know (previous link here was 2-nice-to-know)

Instead we should count the number of exact same title on page and put a running number on them.

This is what I'd imagine we want:

* 0-the-first-heading
* 0-example-heading
* 0-nice-to-know
* 1-example-heading

Then changing the order does not as easily break items unless exact same headings change order

* 0-the-first-heading
* 0-example-heading
* 1-example-heading
* 0-nice-to-know